tiny_dnn/layers/layer.h

```diff
@@ -66,9 +68,13 @@ class layer : public node {
       out_channels_(out_type.size()),
       in_type_(in_type),
       out_type_(out_type) {
+    // todo (karandesai) : remove these after parameter integration
     weight_init_ = std::make_shared<weight_init::xavier>();
     bias_init_   = std::make_shared<weight_init::constant>();
-    trainable_   = true;
+
+    weight_init_f_ = std::make_shared<parameter_init::xavier>();
```

Comment:
I think by default we should just initialize parameters with vanilla uniform_sampling.

Xavier initialization is just one of the many different extensions of uniform sampling that somehow works for CNN. Since we are not just focusing on CNN, this could lead to errors during training other types of networks. @prlz77 might give us a more theoretical understanding about that ?